internal/kafka/kafka.go

```diff
@@ -206,25 +206,29 @@ func (b *kafkaBroker) Consume(ctx context.Context, topic string, since time.Time
 				msg.Key, msg.Topic, msg.Offset, msg.Partition)
 
 			// build new context - identity and trace id
-			ctx, err = identity.WithIdentityFrom64(ctx, header("x-rh-identity", msg.Headers))
-			if err != nil {
-				logger.Trace().Msgf("Could not extract identity from context to Kafka message: %s", err)
+			newLogger := logger.With().Str("msg_id", random.TraceID().String())
+			newCtx, err := identity.WithIdentityFrom64(ctx, header("X-RH-Identity", msg.Headers))
+			if err == nil {
+				id := identity.Identity(newCtx)
+
+				traceId := trace.SpanFromContext(ctx).SpanContext().TraceID()
+				if !traceId.IsValid() {
+					traceId = random.TraceID()
+				}
+				newCtx = logging.WithTraceId(newCtx, traceId.String())
+
+				newLogger = newLogger.
+					Str("trace_id", traceId.String()).
+					Str("account_number", id.Identity.AccountNumber).
+					Str("org_id", id.Identity.OrgID)
+				newCtx = newLogger.Logger().WithContext(newCtx)
+			} else {
+				errLogger := newLogger.Logger()
+				errLogger.Warn().Err(err).Msgf("Could not extract identity from context to Kafka message")
+				newCtx = errLogger.WithContext(newCtx)
```

Comment:
If I understood correctly, you think the issue is because we did not create a new context. 
Why do you change it to err == nil? I would rather be consistent and keep the change like that: 

```suggestion
newLogger := logger.With().Str("msg_id", random.TraceID().String())
newCtx, err := identity.WithIdentityFrom64(ctx, header("X-RH-Identity", msg.Headers))
if err != nil {
errLogger := newLogger.Logger()
				errLogger.Warn().Err(err).Msgf("Could not extract identity from context to Kafka message")
				newCtx = errLogger.WithContext(newCtx)
} else {
		id := identity.Identity(newCtx)

				traceId := trace.SpanFromContext(ctx).SpanContext().TraceID()
				if !traceId.IsValid() {
					traceId = random.TraceID()
				}
				newCtx = logging.WithTraceId(newCtx, traceId.String())

				newLogger = newLogger.
					Str("trace_id", traceId.String()).
					Str("account_number", id.Identity.AccountNumber).
					Str("org_id", id.Identity.OrgID)
				newCtx = newLogger.Logger().WithContext(newCtx)
}
			handler(newCtx, NewMessageFromKafka(&msg))
```

Reply:
It is just a nit, but I find the `err != nil` syntax as a convention and maybe more readable.

Reply:
The problem is that the code overwrites `ctx` variable which becomes a shared variable that is passed over into several goroutines. The solution is to make copies.

The if statement change is just a cleanup - previous behavior when there was no identity was also incorrect - it would only log a warning and then continue trying to read identity while now it correctly only read the identity when it is present.

Reply:
but we could flip the if/else code branches, right?

Reply:
I don’t understand what for, please show me full code which would be more readable than this. My goal is:

* When there is no identity available, log a warning into the newly created logger (which has msg_id field)
* Otherwise extract identity, extract trace id and create a new logger with all the fields.
* Call the message handler with either one or the other newly created context.

Honestly, I tried to swap if statements and it looks the same, I cannot drop `else` or at least I can't figure out how that would be more readable.

Reply:
I did not mean to drop the else, I wrote a suggestion with the same changes you have done, just by switching the branches. 
Just said we usually use err != nil.

Reply:
Let me rebase I need to push anyways to trigger tests again, swapped.